src/serlio/modifiers/PRTModifierAction.cpp

```diff
@@ -810,8 +854,15 @@ MStatus PRTModifierAction::createNodeAttributes(const RuleAttributeSet& ruleAttr
 					const wchar_t* anName = an->getName();
 					if (std::wcscmp(anName, ANNOT_ENUM) == 0)
 						return {AttributeTrait::ENUM, {{}, an}};
-					else if (std::wcscmp(anName, ANNOT_RANGE) == 0)
-						return {AttributeTrait::RANGE, {{}, an}};
+					else if (std::wcscmp(anName, ANNOT_RANGE) == 0) {
+						const RangeType annotationRangeType = GetRangeType(an);
+						if (annotationRangeType == RangeType::ENUM)
+							return {AttributeTrait::ENUM, {{}, an}};
+						else if (annotationRangeType == RangeType::RANGE)
+							return {AttributeTrait::RANGE, {{}, an}};
+						else
+							return {AttributeTrait::PLAIN, {}};

```

Comment:
I would prefer to handle enums with `switch` (the compiler will then complain if not all enum items are handled)

Reply:
switched to switch ;)